src/v/datalake/translation/partition_translator.cc

```diff
@@ -411,7 +419,9 @@ ss::future<> partition_translator::translate_until_stopped() {
           [&needs_jitter] { needs_jitter = true; });
         // Clear the flag as it is a one-shot request, and since
         // translate_until_stopped can be restarted, for example if this
-        // workloop throws.
+        // workloop throws. This avoids clearing the flag before running
+        // run_one_translation_iteration which uses the flag to silence the
+        // out-of-memory warning message.
         auto clear_finish_request = ss::defer(
           [this] { _finish_translation_requested = false; });
```

Comment:
If we just made stop_translator() accept an exception, and used some new translation_stop_requested_exception, could we avoid the additional orchestration of this _finish_translation_requested? It seems a bit more explicit and easier to follow IMO, though maybe it misses out on something that the current impl handles?

Reply:
yes i think so. the catch is that we end up having to change the semantics of stop_translation such that it does one thing for running translator and another thing for a idle translator. that's not a problem per se, i just didn't want to rock the boat too much. happy to prototype that tho it should be quick

Reply:
Ahh got it, makes sense. In that case, I don't mind it as is (though definitely open to the change if it does end up seeming more straightforward)